.github/workflows/on_build_do_publish_image.yml

```diff
@@ -0,0 +1,70 @@
+name: Publish Image
+on:
+   workflow_run:
+      workflows: [Build]
+      types: [completed]
+env:
+   IMAGE_NAME: ${{ vars.QUAY_IMAGE_NAME  || 'quay.io/infinispan-test/server' }}
+
+concurrency:
+  # Only cancel jobs for PR updates
+  group: ci-publish-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  get-info:
+     name: Get Origin Workflow Info
+     runs-on: ubuntu-latest
+     outputs:
+        source-head-sha: ${{ github.event.workflow_run.head_sha }}
+        pull-request-number: ${{ steps.workflow-run-info.outputs.pull-request-number }}
+        sourceEvent: ${{ github.event.workflow_run.event }}
+        image-tag: ${{ steps.workflow-run-info.outputs.image-tag }}
+        publish-image: ${{ steps.publish-image-info.outputs.publish-image }}
+     steps:
+        - uses: actions/checkout@v4
+
+        - id: workflow-run-info
+          uses: ./.github/actions/get-origin-info
+          with:
+           head-sha: ${{ github.event.workflow_run.head_sha }}
+           head-branch: ${{ github.event.workflow_run.head_branch }}
+           event: ${{ github.event.workflow_run.event }}
+
+        - if: ${{ github.event.workflow_run.event == 'pull_request' }}
+          name: Get Labels Action
+          uses: snnaplab/get-labels-action@v1.0.1
+          with:
+            number: ${{ steps.workflow-run-info.outputs.pull-request-number }}
+
+        - if: ${{ github.event.workflow_run.event == 'pull_request' && contains(fromJSON(env.LABELS), 'Image Required') }}
+          id: publish-image-info
+          run: echo "publish-image=true" >> $GITHUB_OUTPUT
+
+  get-image-name:
```

Comment:
Can't this just be a step in the `get-info` job?

Reply:
actually yes